sdk/src/mesh_dp.c

```diff
@@ -0,0 +1,779 @@
+
+    if (!cfg)
+        return -MESH_ERR_BAD_CONFIG_PTR;
+
+    conn_ctx = (MeshConnectionContext *)conn;
+    conn_ctx->cfg.conn_type = MESH_CONN_TYPE_ST2110;
+    mesh_memcpy(&conn_ctx->cfg.conn.st2110, cfg, sizeof(MeshConfig_ST2110));
+
+    return 0;
+}
+
+/**
+ * Apply configuration to setup RDMA connection via Media Proxy
+ */
+int mesh_apply_connection_config_rdma(MeshConnection *conn, MeshConfig_RDMA *cfg)
+{
+    MeshConnectionContext *conn_ctx;
+
+    if (!conn)
+        return -MESH_ERR_BAD_CONN_PTR;
+
+    if (!cfg)
+        return -MESH_ERR_BAD_CONFIG_PTR;
+
+    conn_ctx = (MeshConnectionContext *)conn;
+    conn_ctx->cfg.conn_type = MESH_CONN_TYPE_RDMA;
+    mesh_memcpy(&conn_ctx->cfg.conn.rdma, cfg, sizeof(MeshConfig_RDMA));
+
+    return 0;
+}
+
+/**
+ * Apply configuration to setup connection payload for Video frames
+ */
+int mesh_apply_connection_config_video(MeshConnection *conn, MeshConfig_Video *cfg)
+{
+    MeshConnectionContext *conn_ctx;
+
+    if (!conn)
+        return -MESH_ERR_BAD_CONN_PTR;
+
+    if (!cfg)
+        return -MESH_ERR_BAD_CONFIG_PTR;
+
+    conn_ctx = (MeshConnectionContext *)conn;
+    conn_ctx->cfg.payload_type = MESH_PAYLOAD_TYPE_VIDEO;
+    mesh_memcpy(&conn_ctx->cfg.payload.video, cfg, sizeof(MeshConfig_Video));
+
+    return 0;
+}
+
+/**
+ * Apply configuration to setup connection payload for Audio packets
+ */
+int mesh_apply_connection_config_audio(MeshConnection *conn, MeshConfig_Audio *cfg)
+{
+    MeshConnectionContext *conn_ctx;
+
+    if (!conn)
+        return -MESH_ERR_BAD_CONN_PTR;
+
+    if (!cfg)
+        return -MESH_ERR_BAD_CONFIG_PTR;
+
+    conn_ctx = (MeshConnectionContext *)conn;
+    conn_ctx->cfg.payload_type = MESH_PAYLOAD_TYPE_AUDIO;
+    mesh_memcpy(&conn_ctx->cfg.payload.audio, cfg, sizeof(MeshConfig_Audio));
+
+    return 0;
+}
+
+/**
+ * Parse payload configuration
+ */
+int mesh_parse_payload_config(MeshConnectionContext *conn, mcm_conn_param *param)
+{
+    /**
+     * Parse video payload parameters
+     */
+    if (conn->cfg.payload_type == MESH_PAYLOAD_TYPE_VIDEO) {
+
+        switch (conn->cfg.payload.video.pixel_format) {
+        case MESH_VIDEO_PIXEL_FORMAT_NV12:
+            param->pix_fmt = PIX_FMT_NV12;
+            break;            
+        case MESH_VIDEO_PIXEL_FORMAT_YUV422P:
+            param->pix_fmt = PIX_FMT_YUV422P;
+            break;            
+        case MESH_VIDEO_PIXEL_FORMAT_YUV422P10LE:
+            param->pix_fmt = PIX_FMT_YUV422P_10BIT_LE;
+            break;
+        case MESH_VIDEO_PIXEL_FORMAT_YUV444P10LE:
+            param->pix_fmt = PIX_FMT_YUV444P_10BIT_LE;
+            break;
+        case MESH_VIDEO_PIXEL_FORMAT_RGB8:
+            param->pix_fmt = PIX_FMT_RGB8;
+            break;
+        default:
+            return -MESH_ERR_CONN_CONFIG_INVAL;
+        }
+
+        param->payload_args.video_args.pix_fmt = param->pix_fmt;
+        param->payload_args.video_args.width   = conn->cfg.payload.video.width;
```

Comment:
it won't work for RDMA as it currently needs payload_args.rdma_args filled. As agreed, we can change rdma connection to take video_args, but it has to be done..

Reply:
Good catch. Could you add a suggestion in this comment on what fields you'd like to populate?

Reply:
Here there is nothing to add, but we have to interpret those parameters differently in [ProxyContext::RxStart_rdma](https://github.com/OpenVisualCloud/Media-Communications-Mesh/blob/main/media-proxy/src/proxy_context.cc#L802) and  [ProxyContext::TxStart_rdma](https://github.com/OpenVisualCloud/Media-Communications-Mesh/blob/main/media-proxy/src/proxy_context.cc#L998)

Reply:
Fixed in eae8b083e0210567994eb379b5e228742451c44b.
@Sakoram please review.